llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -274,24 +397,41 @@ int PostPipeliner::mostUrgent() {
   return Best;
 }
 
-bool PostPipeliner::scheduleFirstIteration() {
+void PostPipeliner::resetSchedule(bool FullReset) {
+  Scoreboard.clear();
+  for (int K = 0; K < NTotalInstrs; K++) {
+    auto &N = Info[K];
+    N.reset(FullReset);
+    if (K < NInstr) {
+      N.Earliest = N.StaticEarliest;
+      N.Latest = N.StaticLatest;
+    }
```

Comment:
Does that imply that nodes outside of the first iteration never get an `Earliest` or `Latest`? Should we make that explicit with an `optional`?

Reply:
They get a default 0, -1 from reset, which are sane values, not sentinels. Earliest of the second iteration is used to represent LCD latencies, but this is propagated back to the first iteration.